File: SEIST/server/python/verify_release.py
from __future__ import annotations

import hashlib
import json
import sys
from pathlib import Path


def main() -> int:
    root = Path(sys.argv[1]) if len(sys.argv) > 1 else Path("dist")
    package_path = root / "seis-static.zip"
    manifest_path = root / "server-upload-manifest.json"

    if not package_path.exists() or not manifest_path.exists():
        print("missing release package or manifest", file=sys.stderr)
        return 1

    manifest = json.loads(manifest_path.read_text(encoding="utf-8"))
    digest = hashlib.sha256(package_path.read_bytes()).hexdigest()

    if digest != manifest.get("sha256"):
        print("sha256 mismatch", file=sys.stderr)
        print(f"expected: {manifest.get('sha256')}", file=sys.stderr)
        print(f"actual:   {digest}", file=sys.stderr)
        return 1

    print(json.dumps({
        "ok": True,
        "package": str(package_path),
        "sha256": digest,
        "bytes": package_path.stat().st_size,
    }, indent=2))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: SEIST/polyglot/ruby/verify_release.rb
require "digest"
require "json"

root = ARGV[0] || "dist"
package_path = File.join(root, "seis-static.zip")
manifest_path = File.join(root, "server-upload-manifest.json")

abort("missing package") unless File.file?(package_path)
abort("missing manifest") unless File.file?(manifest_path)

manifest = JSON.parse(File.read(manifest_path))
digest = Digest::SHA256.file(package_path).hexdigest
abort("sha256 mismatch") unless digest == manifest["sha256"]

puts JSON.pretty_generate({
  ok: true,
  package: package_path,
  sha256: digest,
  bytes: File.size(package_path)
})

